Kudu.Services/FetchHelpers/OneDriveHelper.cs

```diff
@@ -0,0 +1,517 @@
+        {
+            if (Logger != null)
+            {
+                Logger.Log(format, args);
+            }
+        }
+
+        private void LogException(Exception ex)
+        {
+            if (Logger != null)
+            {
+                Logger.Log(ex);
+            }
+        }
+
+        private void TraceMessage(string format, params object[] args)
+        {
+            lock (_tracer)
+            {
+                _tracer.Trace(format, args);
+            }
+        }
+
+        private HttpClient CreateHttpClient(string accessToken)
+        {
+            HttpClient client = new HttpClient();
+            client.MaxResponseContentBufferSize = 1024 * 1024 * 10;
+            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(JsonMediaType));
+            if (!client.DefaultRequestHeaders.Contains(UserAgentHeader))
+            {
+                client.DefaultRequestHeaders.Add(UserAgentHeader, MicrosoftAzure);
+            }
+
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("bearer", accessToken);
+
+            return client;
+        }
+
+        private async Task<T> ProcessResponse<T>(string operation, HttpResponseMessage response)
+        {
+            string content = response.Content != null ? await response.Content.ReadAsStringAsync() : null;
+            if (response.IsSuccessStatusCode)
+            {
+                var serializer = new JavaScriptSerializer();
+                return serializer.Deserialize<T>(content);
+            }
+
+            throw new Exception(string.Format(CultureInfo.InvariantCulture, "'{0}' : '{1}', {2}", response.StatusCode, operation, content));
+        }
+
+        private static string GetOneDriveRootUri(string requestUri)
+        {
+            // should come in as https://api.onedrive.com/v1.0/drive/special/approot:/{folder name}
+            var uriTokens = requestUri.Split(new string[] { "/special/" }, StringSplitOptions.None);
+            if (uriTokens.Length != 2)
+            {
+                throw new ArgumentException(string.Format(CultureInfo.CurrentUICulture, Resources.OneDriveInvalidRequestUri, requestUri));
+            }
+
+            return uriTokens[0];
+        }
+
+        private async Task<string> GetItemUri(string accessToken, string requestUri, string rootUri)
+        {
+            using (_tracer.Step("Converting {0} to identify by id", requestUri))
+            {
+                using (var client = CreateHttpClient(accessToken))
+                using (var response = await client.GetAsync(requestUri))
+                {
+                    var item = await ProcessResponse<OneDriveModel.OneDriveItem>("GetItemUri", response);
+
+                    // should return as https://api.onedrive.com/v1.0/drive/items/{folder id}
+                    return string.Format(CultureInfo.InvariantCulture, "{0}/items/{1}", rootUri, item.id);
```

Comment:
Here i convert the given URL to new URL that with item ID. Alternatively we can ask caller to pass in a URL template.